pkg/obfuscator/ip.go

```diff
@@ -35,9 +36,12 @@ type ipGenerator struct {
 	template   string
 	obfuscated string
 	count      int
+	lock       sync.Mutex
 }
 
 func (g *ipGenerator) generateConsistent() string {
+	g.lock.Lock()
+	defer g.lock.Unlock()
```

Comment:
I can't comment down below because that's out of the diff, but I do believe there is a race with the reporter:

```
				case schema.ObfuscateReplacementTypeConsistent:
					if replacement = o.GetReplacement(m); replacement == "" {
						replacement = gen.generateConsistent()
					}
				}
				output = strings.ReplaceAll(output, m, replacement)
				o.ReportReplacement(m, replacement)
```

you can run into a scenario where you find there was no replacement in two threads, generate two different replacements and then report both later which will exit with an error while reporting the second one.

Go should pick this up with a race detector actually :octopus:

Reply:
Indeed, I've made some changes which should fix the issue.